### PR TITLE
[BYOC-DNNL] add support for more ops and fusion patterns

### PR DESCRIPTION
**Pull request by crazydemo**
Enabled OP List:

- Convolution 1D-3D
- Deconvolution 2D-3D
- Pooling 2D-3D (avg / max)
- Elementwise OPs (abs, clip, exp, log, sqrt, round, logsumexp, relu, leaky_relu, tanh, sigmoid, softmax)

### CODE REVIEW

**Comment by tmoreau89** [comment]
CC @apeskov and @mbs-octoml

**Comment by crazydemo** [comment]
Hi, @apeskov and @mbs-octoml. Could you please take some time to review this PR? Look forward to your comments.

**Comment by mbs-octoml** [comment]
@mikepapadim can you take a look at a this one, thanks.

**Comment by crazydemo** [comment]
> > @mikepapadim can you take a look at a this one, thanks.
> 
> I ll take a look!

Hi @mikepapadim, do you have any comments?

**Comment by crazydemo** [comment]
@tmoreau89 @mbs-octoml @mikepapadim @masahi Could you please help me review this PR?

**Comment by masahi** [comment]
I can merge this after you pass CI.

**Comment by crazydemo** [comment]
> I can merge this after you pass CI.
Thank you! I will check!

**Comment by crazydemo** [comment]
Hi @masahi , I wonder which version of OneDNN is used for CI.
I have one build error, which says `'eltwise_logsigmoid' is not a member of 'dnnl::algorithm'`, and this algo is supported by OneDNNv1.8 or the later version.
Could you help me skip this check?

**Comment by masahi** [comment]
https://github.com/apache/tvm/blob/813136401a11a49d6c15e6013c34dd822a5c4ff6/docker/install/ubuntu_install_dnnl.sh#L24

Probably too old, we should update that first.

**Comment by crazydemo** [comment]
I will try to update to `OneDNN V2.2`.

**Comment by masahi** [comment]
Note that changing the docker script won't immediately update the CI environment.You need to send that change separately first like https://github.com/apache/tvm/pull/10179. I can take care of actually update the ci image.

**Comment by crazydemo** [comment]
> Note that changing the docker script won't immediately update the CI environment (so the build will fail again).You need to send that change separately first like #10179. I can take care of actually updating the ci image.

Do I need to submit an RFC, or just submit an independent PR?

**Comment by masahi** [comment]
No need for another rfc.

**Comment by vpirogov** [comment]
The latest oneDNN release is [v2.5.2](https://github.com/oneapi-src/oneDNN/releases/tag/v2.5.2)

**Comment by crazydemo** [comment]
> The latest oneDNN release is [v2.5.2](https://github.com/oneapi-src/oneDNN/releases/tag/v2.5.2)

Yes, the latest release version is v2.5.2, but only v2.2 has [dnnl_lnx_2.2.0_cpu_gomp.tgz](https://github.com/oneapi-src/oneDNN/releases/download/v2.2/dnnl_lnx_2.2.0_cpu_gomp.tgz)

**Comment by crazydemo** [comment]
The PR, updating CI has been submitted [#10266](https://github.com/apache/tvm/pull/10266) @masahi

**Comment by vpirogov** [comment]
@crazydemo, right, we stopped publishing binary releases on Github after v2.2. Binaries for later versions are available on [conda-forge](https://anaconda.org/conda-forge/onednn/files).

**Comment by crazydemo** [comment]
@vpirogov Thank you so much! I will update the CI to the latest version.

**Comment by crazydemo** [comment]
hi @masahi , how can I restart the ci check?

**Comment by masahi** [comment]
So this is how it works. 

* The new CI image was created 3 hours ago in https://hub.docker.com/r/tlcpackstaging/ci_cpu/tags that contains the new dnnl change. I need to test this image in the staging branch, which I just started at https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/ci-docker-staging/217/pipeline. 
* After that CI job passes, I'll open a PR to update `Jenkinsfile` to point to the new cpu image.
* After that PR is merged, you can resume your work in this PR.

CI update is an involved process. So please wait until at least tomorrow.

**Comment by crazydemo** [comment]
Got it! Thank you for all your help and explanation!

**Comment by masahi** [comment]
@crazydemo it's ready now.

File: python/tvm/relay/op/contrib/dnnl.py
Comment (masahi) [inline python/tvm/relay/op/contrib/dnnl.py:102]: We don't use `deconv` terminology. Since this is a pattern for Relay graph, please use `conv2d_transpose`.

**Review by masahi — APPROVED** [review]
LGTM except one minor nit. Can be address in a later PR if more is coming.

File: python/tvm/relay/op/contrib/dnnl.py
Comment (crazydemo) [reply python/tvm/relay/op/contrib/dnnl.py:102]: Thank you for your help. I will change it into `conv2d_transpose` in the next PR.

**Comment by masahi** [comment]
@crazydemo please look at the CI issues.

**Pull request merged**
